tycho-core/src/main/java/org/eclipse/tycho/p2resolver/P2ResolverImpl.java

```diff
@@ -227,9 +227,7 @@ protected P2ResolutionResult resolveDependencies(Collection<IInstallableUnit> ro
             if (project != null && p2ResolverFactoryImpl != null && pomDependencies != PomDependencies.ignore) {
                 TargetPlatformConfiguration configuration = p2ResolverFactoryImpl.getProjectManager()
                         .getTargetPlatformConfiguration(project);
-                if (!configuration.isRequireEagerResolve()) {
-                    data.setAdditionalUnitStore(p2ResolverFactoryImpl.getPomUnits().createPomQueryable(project));
-                }
+                data.setAdditionalUnitStore(p2ResolverFactoryImpl.getPomUnits().createPomQueryable(project));
```

Comment:
This is only valid in lazy mode, because in eager mode all dependencies are already resolved beforehand and can't be read from the pom (again).